portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -64,4 +74,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String json = new Gson().toJson(commentList);
     response.getWriter().println(json);
   }
+
+  private String getUsername(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("user").setFilter(
+        new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+
+    if (entity == null) {
+      return null;
+    }
+
+    String nickname = (String) entity.getProperty("username");
+    return nickname;
```

Comment:
Could inline this all to
```suggestion
    return entity == null ? null : (String) entity.getProperty("username");
```

Reply:
This inline throws an error since I believe that the property "username" of entity is tried even if entity is null

Therefore entity==null has to be checked beforehand... Although I may be missing something

Reply:
Nvm got it!